docs/deployment/kubernetes.md

```diff
@@ -0,0 +1,34 @@
+# Deploying the Node Controller Manager into a Kubernetes cluster
+
+As already mentioned, the Node Controller Manager is designed to run as controller in a Kubernetes cluster. The existing source code can be compiled and tested on a local machine as described in [Setting up a local development environment](../development/local_setup.md). Once the implementation is stabilized, you can deploy this onto your cluster using the steps described below.
+
+## Prepare the cluster
+
+- Connect to the remote kubernetes cluster where you plan to deploy the Node Controller Manager using the kubectl. Make the KUBECONFIG environment variable point to the yaml file containing your cluster info.
+- Now, create the required CRDs on the remote cluster using the following command,
+```bash
+$ kubectl apply -f kubernetes/crds.yaml
+```
+
+## Build the Docker image
+
+:warning: Modify the default docker registry with your own, modify the `Makefile` to point it to your own registry.
+
+- Build the existing source code into a binary. The compiled binary is present at `bin/node-controller-manager`
+```bash
+$ make build
+```
+- Build docker image from latest compliled binary
+```bash
+$ make docker-image
+```
+- Push the last created docker image onto the online docker registery. 
+```bash
+$ make push
+```
+
+- Now you can use this docker image on the online registry to create a deployment in your cluster. A sample development [file is given at](/kubernetes/deployment.yaml). Make sure that your deployment is run using a flag to your cluster's [Kubeconfig](https://kubernetes.io/docs/tasks/access-application-cluster/authenticate-across-clusters-kubeconfig/)
```

Comment:
Now you can deploy this docker image to your cluster. A sample deployment file ...

Reply:
Done.
https://github.com/gardener/node-controller-manager/commit/95c68a4c3f483a5f770c7de61f12b25e5411f3c0#diff-cf6b5c064a3b7eca6025325785cbbea7R30